tarogarden/seedling.go

```diff
@@ -124,6 +129,47 @@ func (c Seedling) validateFields() error {
 	return nil
 }
 
+// validateGroupKey attempts to validate that the non-zero group key provided
+// with a seedling is owned by the daemon and can be used with this seedling.
+func (c Seedling) validateGroupKey(groups []*asset.AssetGroup) (
+	*asset.AssetGroup, error) {
+
+	// Scan for a group that matches our seedling group key.
```

Comment:
See comment above re just fetching this directly instead of doing an in-mem scan. I like the validation here though, just to make sure the user actually knows what they're doing.